### ISSUE TITLE
ClassCastException when generating sample XML from XSD

### ISSUE THREAD

**Issue by emu42** [santhosh-tekuri/jlibs#71]
Hello,

I am currently trying to get a sample XML document from a fairly complex set of XSDs. Now I am facing this exception:
```
Caused by: java.lang.ClassCastException: org.apache.xerces.impl.dv.xs.XSSimpleTypeDecl cannot be cast to org.apache.xerces.xs.XSComplexTypeDefinition
        at jlibs.xml.xsd.XSUtil.getSubTypes(XSUtil.java:70)
        at jlibs.xml.xsd.XSInstance$XSSampleNavigator.process(XSInstance.java:170)

```
I did some debugging and apparently this exception is thrown when processing the type base64Binary from http://www.w3.org/2005/05/xmlmime (which is included in my set of XSDs):
```
  <xs:complexType name="base64Binary">
    <xs:simpleContent>
        <xs:extension base="xs:base64Binary">
            <xs:attribute ref="xmime:contentType"/>
        </xs:extension>
    </xs:simpleContent>
  </xs:complexType>
```

Any suggestions on how to get past this?

**Comment by emu42** [comment]
Altering the loop in XSUtil.java (lines 81-83) seems to do the trick:
```
                    while (complexItem!=anyType && complexItem!=complexType && complexItem.getBaseType() instanceof XSComplexTypeDefinition){
                        complexItem = (XSComplexTypeDefinition)complexItem.getBaseType();
                    };
```
The base type is now checked before the cast.

**Comment by santhosh-tekuri** [comment]
fixed in 3dc2b700a1bc183e6339203aab299ea2e265e08f
